### DISCUSSION TITLE
Windows authentication (SPNEGO / KERBEROS)

### DISCUSSION THREAD

**Discussion by brunogenovese** [Kong/insomnia#4011] [Ideas]
Following an old Insomnia thread and looking at the list of supported authentication methods I get the impression that Insomnia does not support windows authentication (even though it does support the obsolete and security-flawed NTLM protocol).

I strongly recommend adding SPNEGO / KERBEROS (or whatever is the *current* windows authentication standard) support.

Reasons:

- Pretty much all modern browsers support SPNEGO authentication, thus Insomnia should support it too.

- NTLM (its predecessor) is obsolete due to security concerns.  I am not positive but I believe Windows 10 no longer supports NTLM.

- Modern security expectations are that passwords no longer be stored in files, not even encrypted.  Thus where applications could before use their own service username/password stored in a properties file to access certain resources, that is no longer acceptable.  Windows authentication is thus the ideal and most secure way to access a REST service when called from an application using its own credentials - as the password would simply be inherited from the windows "services" definition that launches the client app (i.e.: what Tomcat runs as).

**Comment by ghost** [comment]
I agree with this feedback. I am currently developing an internal company API and would really benefit from having Kerberos authentication.

**Comment by brunogenovese** [comment]
UPDATE:  

I don't know how accurate this is, but I've been told by a co-worker that played with it that it is possible that what is marked as NTLM in Insomnia (as well as Postman) is really the full thing, as in it might first be trying SPNEGO/Kerberos, then the old NTLM if enabled, and then asking for username/password. 

I will post back once I have a chance to test it myself.

**Reply by brunogenovese** [reply]
UPDATE:  We did some tests with both Insomnia and Postman.   We were able to successfully do Windows Authentication (labeled NTLM) using Postman, but not Insomnia.  

The Insomnia developers might be able to see what makes it work in Postman and implement the same way.

**Reply by de846** [reply]
> UPDATE: We did some tests with both Insomnia and Postman. We were able to successfully do Windows Authentication (labeled NTLM) using Postman, but not Insomnia.

I'm surprised that worked. It would depend on the service you tried to authenticate to, as I suspect that service supports NTLM and Kerberos authentication. Did you provide a username and password with your Postman HTTP request when it was set to NTLM?

I tested this in Postman 8.x and 9.x with a basic GET request against a service that only supports Kerberos, and it expectedly responds with  the header `"WWW-Authenticate": "Negotiate"`.

I'd love this (Kerberos) support in Insomnia. As of now, the Postman Chrome App is the only solution available to me at work while my Insomnia installation is growing dusty.
